### DISCUSSION TITLE
I get an error when trying to open an ipybn file in VScode. - 'Expected a JSON object, array or literal.'

### DISCUSSION THREAD

**Discussion by That-Ruben** [microsoft/vscode-jupyter#12656] [Questions and Answers] [answered]
I'm probably not understanding something very basic, but there's no issue with creating, saving or loading new ipybn files. 

The error occurs with this file. (I can't attach ipybn so I just saved it as .txt)

[Untitled-1.txt](https://github.com/microsoft/vscode-jupyter/files/10513497/Untitled-1.txt)
It basically looks like an html page to me, but I don't know why someone would save that with an ipybn extension, so I'm assuming that this is some variant of ipybn files that I'm not equipped to load properly

I'm all googled out, so I'm hoping someone can point out my mistake

**Comment by DonJayamanne** [comment, marked as answer]
Yes as you have identified the ipynb file is invalid. Now sure where you got this file from, but its not a valid file.
Hence the issue isn't with vscode or the jupyter extension, but the fact that you have an invalid (currupted) file that you are trying to open as a Jupyter notebook  file.

My suggestion is to downlod the file into an ipynb format and ensure it has JSON (not sure how you downloaded/saved/got the ipynb file). but i'd look at that process.

**Comment by That-Ruben** [comment]
Sorry for the late reply. Thanks for your answer. 

I downloaded the file from a github page that belongs to someone who creates python tutorial videos on Youtube. So there's a ipynb file to go alone with each video

I'm not sure what the rules are on the forums here so I didn't post a link to it. The files are listed as ipynb files on the github page, and I just used right click + save as. 

Anyway, it wasn't really anything special, so we can consider this case closed.

**Comment by techUdayMungalpara** [comment]
## i also facing the same issue 
![Screenshot from 2024-01-29 16-03-42](https://github.com/microsoft/vscode-jupyter/assets/105286180/90f23e21-e514-46f0-a92e-966186d506a5)
Description:	Ubuntu 22.04.3 LTS
Release:	22.04
![Screenshot from 2024-01-29 16-06-15](https://github.com/microsoft/vscode-jupyter/assets/105286180/e27bb2f8-08a5-4bbe-9f3c-85408bffaa5b)

**Comment by vmehera123** [comment]
Same error even with new ipynb file. 
Re-install jupyter didn't help.

Version: 1.86.0
Commit: 05047486b6df5eb8d44b2ecd70ea3bdf775fd937
Date: 2024-01-31T10:29:11.933Z
Electron: 27.2.3
ElectronBuildId: 26495564
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Darwin arm64 23.3.0

**Comment by vmehera123** [comment]
Any updates?

**Comment by DonJayamanne** [comment]
This happens when the  notebook (json) file is not well formed.
Ie you will not be able to open the same file in Jupyter lab/notebook either

It's not a problem with vscode, but the filte itself

See comment here https://github.com/microsoft/vscode-jupyter/discussions/12656#discussioncomment-4792629

**Comment by vmehera123** [comment]
@DonJayamanne No, it's not! Check out my [comment](https://github.com/microsoft/vscode-jupyter/discussions/12656#discussioncomment-8353901) . I'm able to open ipynb files within 1.81 version, but get and error, while doing so in 1.86

**Reply by DonJayamanne** [reply]
You have an old version of vscode, please install the latest version and try again

**Reply by DonJayamanne** [reply]
Jupyter extension isn't required to open ipynb (notebook) files. It's handled natively by vscode
